### PR TITLE
Add defaultMaxRecoveryAttempts to RequestDefaults and use for maxRecoveryAttempts in Request.

### PR DESCRIPTION
**Pull request by earlgaspard**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -78,6 +78,10 @@
 [Earl Gaspard](https://github.com/earlgaspard)
 [#135](https://github.com/BottleRocketStudios/iOS-Hyperspace/pull/135)
 
+* Add `defaultMaxRecoveryAttempts` to `RequestDefaults` and use for `maxRecoveryAttempts` in `Request`.
```
Comment (br-tyler-milner) [inline CHANGELOG.md:81]: Might be good to also note that we changed the default from "unlimited retries" to "1 retry".

File: Sources/Hyperspace/Request/Request.swift [hunk 2]
```diff
@@ -38,10 +38,10 @@ public struct Request<Response, Error: TransportFailureRepresentable>: Recoverab
     /// The creation strategy for generating a `URLRequest`. Can be customized by using the `.custom` case.
     public var urlRequestCreationStrategy: URLRequestCreationStrategy = .default
     
-    /// The maximum number of attempts that this operation should make before completely aborting. This value is nil when there is no maximum.
-    public var maxRecoveryAttempts: UInt?
+    /// The maximum number of attempts this operation should make before completely aborting. A value of nil means there is no maximum.
```
Comment (br-tyler-milner) [inline Sources/Hyperspace/Request/Request.swift:41]: Minor: could add backpacks to `nil` for syntax highlighting in the comment.

**Review by br-tyler-milner — APPROVED** [review]

File: Sources/Hyperspace/Request/Request.swift (hunk 2 above)
Comment (earlgaspard) [reply Sources/Hyperspace/Request/Request.swift:41]: Can change.

File: CHANGELOG.md (hunk 1 above)
Comment (earlgaspard) [reply CHANGELOG.md:81]: Ok

**Review by br-tyler-milner — APPROVED** [review]

**Review by tylermilner — APPROVED** [review]

**Review by wmcginty — APPROVED** [review]

**Pull request merged**